src/socket.c

```diff
@@ -1058,26 +1058,24 @@ void clear_socket(D_SOCKET *sock_new, int sock)
 void *lookup_address(void *arg)
 {
   LOOKUP_DATA *lData = (LOOKUP_DATA *) arg;
-  struct hostent *from = 0;
-  struct hostent ent;
-  char buf[16384];
-  int err;
+
+  char hbuf[NI_MAXHOST], sbuf[NI_MAXSERV];
+  int rc;
 
   /* do the lookup and store the result at &from */
-  gethostbyaddr_r(lData->buf, sizeof(lData->buf), AF_INET, &ent, buf, 16384, &from, &err);
+  rc = getnameinfo(lData->sa, lData->sa->sa_len, hbuf, sizeof(hbuf), sbuf, sizeof(sbuf), NI_NAMEREQD);
```

Comment:
It looks like `sa_len` is only found in the BSD definition of the `sockaddr` struct, so this breaks our Linux compatibility. Would you replace the `lData->sa->sa_len` parameter with a `sizeof(lData->sa)`? That seems to bring us cross-platform compatibility.